lib/integrations/awsoidc/eks_enroll_clusters.go

```diff
@@ -0,0 +1,472 @@
+	creds, err := credsProvider.Retrieve(ctx)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	signer := v4.NewSigner()
+	presigned, _, err := signer.PresignHTTP(ctx, creds, r, hashForGetRequests, "sts", region, timestamp)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	kubeToken := awsKubePrefix + base64.RawURLEncoding.EncodeToString([]byte(presigned))
+
+	eksClusterCA, err := base64.StdEncoding.DecodeString(clusterCA)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	restConfig := &rest.Config{
+		Host:        clusterEndpoint,
+		BearerToken: kubeToken,
+		TLSClientConfig: rest.TLSClientConfig{
+			CAData: eksClusterCA,
+		},
+	}
+
+	kubeConfig := genericclioptions.NewConfigFlags(false)
+	kubeConfig.APIServer = &restConfig.Host
+	kubeConfig.BearerToken = &restConfig.BearerToken
+	kubeConfig.WithWrapConfigFn(func(*rest.Config) *rest.Config {
+		return restConfig
+	})
+
+	return kubeConfig, nil
+}
+
+func getHelmActionConfig(kubeConfig genericclioptions.RESTClientGetter, log logrus.FieldLogger) (*action.Configuration, error) {
+	actionConfig := new(action.Configuration)
+	if err := actionConfig.Init(kubeConfig, agentNamespace, "secret", log.WithField("source", "helm").Debugf); err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	return actionConfig, nil
+}
+
+func getHelmSettings() *helmCli.EnvSettings {
+	helmSettings := helmCli.New()
+	helmSettings.RepositoryCache = os.TempDir()
+	helmSettings.SetNamespace(agentNamespace)
+
+	return helmSettings
+}
+
+// checkAgentAlreadyInstalled checks through the Helm if teleport-kube-agent chart was already installed in the EKS cluster.
+func checkAgentAlreadyInstalled(actionConfig *action.Configuration) (bool, error) {
+	listCmd := action.NewList(actionConfig)
+
+	releases, err := listCmd.Run()
+	if err != nil {
+		return false, trace.Wrap(err)
+	}
+	for _, release := range releases {
+		if release.Name == agentName {
+			return true, nil
+		}
+	}
+	return false, nil
+}
+
+// installKubeAgent installs teleport-kube-agent chart to the target EKS cluster.
+func installKubeAgent(ctx context.Context, eksCluster *eksTypes.Cluster, proxyAddr string, actionConfig *action.Configuration, settings *helmCli.EnvSettings, req EnrollEKSClustersRequest) error {
+	installCmd := action.NewInstall(actionConfig)
+	installCmd.RepoURL = agentRepoURL
+	installCmd.Version = teleport.Version
+	if strings.Contains(teleport.Version, "dev") {
+		installCmd.Version = "14.3.0" // For testing during development.
+	}
+
+	chartPath, err := installCmd.LocateChart(agentName, settings)
+	if err != nil {
+		return trace.Wrap(err, "could not locate chart")
+	}
+
+	agentChart, err := loader.Load(chartPath)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	installCmd.ReleaseName = agentName
+	installCmd.Namespace = agentNamespace
+	installCmd.CreateNamespace = true
+	vals := map[string]any{}
+	vals["proxyAddr"] = proxyAddr
+	vals["kubeClusterName"] = aws.ToString(eksCluster.Name)
```

Comment:
should kubeClusterName follow the same structure we use for auto discovery?

Reply:
Changed it to use `ApplyEKSNameSuffix()` - https://github.com/gravitational/teleport/pull/36594/commits/7c8b4a355f447de25172d7dc2aaec666d0df8a44